File: testsuite/basic/src/test/java/io/smallrye/faulttolerance/reuse/async/completionstage/MyFaultTolerance.java
package io.smallrye.faulttolerance.reuse.async.completionstage;

import static java.util.concurrent.CompletableFuture.completedFuture;

import java.util.concurrent.CompletionStage;

import jakarta.enterprise.context.ApplicationScoped;
import jakarta.enterprise.inject.Produces;

import io.smallrye.common.annotation.Identifier;
import io.smallrye.faulttolerance.Types;
import io.smallrye.faulttolerance.api.TypedGuard;

@ApplicationScoped
public class MyFaultTolerance {
    @Produces
    @Identifier("my-fault-tolerance")
    public static final TypedGuard<CompletionStage<String>> GUARD = TypedGuard.create(Types.CS_STRING)
            .withRetry().maxRetries(2).done()
            .withFallback().handler(() -> completedFuture("fallback")).done()
            .build();
}

File: testsuite/basic/src/test/kotlin/io/smallrye/faulttolerance/kotlin/reuse/fallback/typedguard/MyFaultTolerance.kt
package io.smallrye.faulttolerance.kotlin.reuse.fallback.typedguard

import io.smallrye.common.annotation.Identifier
import io.smallrye.faulttolerance.api.TypedGuard
import jakarta.enterprise.context.ApplicationScoped
import jakarta.enterprise.inject.Produces
import java.util.function.Supplier

@ApplicationScoped
object MyFaultTolerance {
    @Produces
    @Identifier("my-fault-tolerance")
    val GUARD = TypedGuard.create(String::class.java)
        .withRetry().maxRetries(2).done()
        .withFallback().handler(Supplier { "fallback" }).done()
        .build()
}
